### DISCUSSION TITLE
Missing route times

### DISCUSSION THREAD

**Discussion by tukusejssirs** [jmcdo29/ogma#1376] [Q&A]
When using Ogma logger as Nest general logger (as described [here](https://github.com/jmcdo29/ogma/tree/main/packages/nestjs-module#putting-it-all-together)), we lose information about time how long it took for `RoutesResolver` and `NestApplication` to do their job (e.g. `+3ms` for `AppController` below).

```ts
// Nest logger
[Nest] 1289180  - 29/08/2022, 18:40:29     LOG [RoutesResolver] AppController {/api}: +3ms
[Nest] 1289180  - 29/08/2022, 18:40:29     LOG [RouterExplorer] Mapped {/api, GET} route +3ms
[Nest] 1289180  - 29/08/2022, 18:40:29     LOG [NestApplication] Nest application successfully started +2ms

// Ogma logger
[2022-08-29T16:41:01.005Z] [INFO]  [host] [test-app] [1290375]  [RoutesResolver] AppController {/api}:
[2022-08-29T16:41:01.008Z] [INFO]  [host] [test-app] [1290375]  [RouterExplorer] Mapped {/api, GET} route
[2022-08-29T16:41:01.009Z] [INFO]  [host] [test-app] [1290375]  [NestApplication] Nest application successfully started
```

Is it possible to add this information somewhere?

**Comment by jmcdo29** [comment]
Currently, no. Ogma doesn't keep an internal time clock like Nest's logger does, so there's no time diff to post. I suppose that's something we could add, but the information is already there with the timestamps that Ogma logs vs the simple timings that Nest's logger uses (notice that Ogma's already includes the millisecond while Nest's does not). 

It is something we _could_ add, but I'm not sure how much value it brings to the logs themselves

**Reply by tukusejssirs** [reply]
I thought that Nest somehow propagates those durations, but it seems like you are telling me that it is not.

As for the milliseconds, yes I like it very much that Ogma uses UTC time string! :+1:

As for the time calculation, I thought that Nest measures how long an operation (approximately) took, therefore calculating difference between two consequent logs isn’t useful at all.

As for the value of this kind of information, I can think of a single one for users: if an operations takes longer that a user thinks it should, they could investigate the issue.

However, if Nest does not share the timings with loggers, I think it might be a good idea to implement this in Nest first.

**Reply by jmcdo29** [reply]
The other thing is that ogma doesn't _extend_ the liver from nest, it just _implements_ the interface. I could look into adding this if there's enough of a request for it though
